src/module.c

```diff
@@ -13983,7 +13983,7 @@ RedisModuleDict *RM_DefragRedisModuleDict(RedisModuleDefragCtx *ctx, RedisModule
     }
 
     while (raxNext(&ri)) {
-        void *newdata = valueCB(ri.data, ri.key, ri.key_len);
+        void *newdata = valueCB(ctx, ri.data, ri.key, ri.key_len);
         if (newdata)
             raxSetData(ri.node, ri.data=newdata);
```

Comment:
maybe we better change the test to actually store data in the dict and defrag it?
it would have mean that we'll realize that we're missing a defrag context in the callback.
and maybe, we'll discover other issues (changing the rax during iteration) :shrug:

Reply:
> I was wondering if we need to add privdata in the callback
I don't see why not. `privdata` is not currently needed for the `dict`s used by `TS`. But there' no harm in future-proofing this against modules that do need it.

Reply:
not arguing against it, but just suggesting to keep it consistent with other similar APIs (if they do or don't have privdata).
technically, since this is single threaded, and the CB is called from an API the module calls (not a hook CB that's used later), the module can always just store that info in a global..